### PR TITLE
quant bn/gn/in: move scale and zp to buffers

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45313 quant bn/gn/in: move scale and zp to buffers**

Summary:

Fixes https://github.com/pytorch/pytorch/issues/43774

Currently the scale and zero_point in these modules is not
saved in the state_dict, so saving them and loading them
back does not work.  Fixing by making them buffers.

Note: not handling loading old state_dicts on purpose as unfortunately
there isn't a way to recover correct scales and zero_points as we did
not save them.

The # of people affected by this should be small,
as most people fuse BNs into preceding layers during quantization, and quantized
BN/IN support is very new.

Test Plan:

```
python test/test_quantization.py TestStaticQuantizedModule
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23919143](https://our.internmc.facebook.com/intern/diff/D23919143)

### CODE REVIEW

File: torch/nn/quantized/modules/normalization.py [hunk 1]
```diff
@@ -120,13 +121,13 @@ def __init__(self, num_features, weight, bias, scale, zero_point,
             num_features, eps, momentum, affine, track_running_stats)
         self.weight = weight
         self.bias = bias
-        self.scale = scale
-        self.zero_point = zero_point
+        self.register_buffer('scale', torch.Tensor(scale))
```
Comment (jerryzh168) [inline torch/nn/quantized/modules/normalization.py:124]: make scale float?

**Review by jerryzh168 — APPROVED** [review]
please make scale Tensor float as well

**Comment by vkuzo** [comment]
> please make scale Tensor float as well

sure, it defaults to float but will add it to clarify intent

**Review by z-a-f — APPROVED** [review]
lgtm

**Comment by vkuzo** [comment]
abandoning, doesn't seem worth it yet to fix all the callsites depending on the existing behavior, we may revisit in the future
